content/en/docs/Configuration/debugging-kiali.md

```diff
@@ -18,25 +18,184 @@ spec:
+* `GraphNamespaces`: generates a namespaces graph
+* `GraphAggregate`: generates a node detail graph
+* `MeshGraph`: generates a mesh graph
+
+#### Show logs of processing times
+
+Kiali collects metrics of its internal systems to track its performance (see the next section, "Metrics"). Many of these metrics use a timer to measure the duration of time that Kiali takes to process some unit of work (for example, the time it takes to generate a graph). Kiali will log these duration times as well as export them to Prometheus. To see what metric timers Kiali is tracking internally, you can do this:
+
+| | |
+|-|-|
+| text: | `grep -o 'timer=[^ ]*' \| sed 's/^timer=//' \| sort -u` |
+| json: | `jq -rR 'fromjson? \| select(.timer) \| .timer' \| sort -u` |
+
+<br/>Note that Kiali will not log times that are under 3 seconds since those are deemed uninteresting and logging them will make the logs "noisy". Prometheus will still collect those metrics, so they are still being recorded.
+
+One timer is especially useful - the timer named "GraphGenerationTime". You can query the log for all the graph generation times like this:
+
+| | |
+|-|-|
+| text: | `grep 'timer=GraphGenerationTime'` |
+| json: | `jq -R 'fromjson? \| select(.timer == "GraphGenerationTime")'` |
+
+<br/>Each log message contains a `duration` attribute - this is the amount of time it took to generate the graph (or parts of the graph). Look at the additional attributes for details on what the timer measured.
+
+Some timers that may be of interest are:
+* `APIProcessingTime`: The time it takes to process an API request in its entirety
+* `GraphGenerationTime`: The time it takes to generate a full graph
+* `GraphAppenderTime`: The time it takes for an appender to decorate a graph
+* `PrometheusProcessingTime`:  The time it takes to run Prometheus queries
+* `TracingProcessingTime`: The time it takes to run Tracing queries
+* `ValidationProcessingTime`: The time it takes to validate a set of Istio configuration resources
+* `SingleValidationProcessingTime`: The time it takes to validate an Istio configuration resource
+* `CheckerProcessingTime`: The time it takes to run a specific validation checker
+
+## Metrics
+
+Kiali has a [metrics endpoint that can be enabled](/docs/configuration/kialis.kiali.io/#.spec.server.observability.metrics.enabled), allowing Prometheus to collect Kiali metrics. You can then use Prometheus (or Kiali itself) to examine and analyze these metrics.
+
+To see the metrics that are currently being emitted by Kiali, you can run the following command which simply parses the metrics endpoint data and outputs all the metrics it finds:
+
+```sh
+curl -s http://<KIALI_HOSTNAME>:9090/metrics | grep -o '^# HELP kiali_.*' | awk '{print $3}'
+```
+
+The Kiali UI itself graphs some of these metrics. In the Kiali UI, navigate to the _Kiali_ workload and select the "Kiali Internal Metrics" tab:
```

Comment:
I'd like to see something similar for how to use Kiali to investigate Kiali's own logs. Unless the logs look bad...

Reply:
the logs will look bad today. We would have to wait for this to be able to show anything: https://github.com/kiali/kiali/issues/7066